### DISCUSSION TITLE
Replacing glfw with SDL?

### DISCUSSION THREAD

**Discussion by yuvaltassa** [google-deepmind/mujoco#396] [Polls]
Hi folks,

We are considering replacing [`glfw`](https://www.glfw.org/) with [`SDL`](https://www.libsdl.org/) as our window and context manager for the built in viewer. This is mostly for internal reasons, we don't expect any usability regressions.

Does anyone have an opinion on this proposed change?

**Comment by cvilas** [comment]
Move to SDL3 when it becomes available?

**Reply by yuvaltassa** [reply]
Last we looked at SDL (2.x I believe) it was a very heavy dependency, while glfw is rather minimal. Is there a particular reason to prefer SDL?

**Reply by cvilas** [reply]
True. GLFW is fine for MuJoCo. I was being a bit selfish here, considering _my_ use cases where MuJoCo is just one component within a larger cross-platform project. 

In such use cases (robotics, for instance), other components exist that require not just window management but also input handling (including joysticks, cameras, audio) and GPU support (for compute intensive algorithms). SDL3 [does well here](https://wiki.libsdl.org/SDL3/NewFeatures) - a single third party dependency that could eliminate dependency on multiple third party libraries (say GLFW3 + GStreamer + etc)

Just built SDL3 from source on my laptop and it seems to be incredibly fast to build. Doesn't feel bloated, although I may be wrong.
